articles/applications/reference/wildcard-subdomains.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Wildcards for Subdomains
+description: Describes wildcards for subdomains function in application configuration.
+toc: true
+topics:
+  - applications
+contentType: reference
+useCase:
+  - build-an-app
+---
+# Wildcards for Subdomains
+
+You can use wildcards for subdomain URL registration in your application configuration in the Dashboard with these fields:
+ 
+* **Allowed Callback URLs**: Set of URLs to which Auth0 is allowed to redirect users after they authenticate.
+* **Allowed Logout URLs**: List of URLs to which you can redirect users after they log out from Auth0.
+* **Allowed Origins (CORS)**: Set of URLs that will be allowed to make requests from JavaScript to Auth0 API (typically used with CORS).
+
+See [Application Settings Best Practices](/best-practices/application-settings) for recommended settings for applications, which suggests to avoid using wildcards for subdomains in application callbacks and allowed origins. If you choose to use them, you can use the star symbol (`*`) as a wildcard for subdomains, but it must be used in accordance with the following rules in order to properly function:
+
+* The protocol of the URL **must** be `http:` or `https:`. `com.example.app://*.example.com` will not work.
+
+* The wildcard **must** be located within the hostname component of the URL. Star symbols (`*`) located outside of the hostname, such as in the path or query, will not be treated as wildcards. `https://example.com/*` will not be treated as having a wildcard.
+
+* The wildcard **must** be located in a subdomain within the hostname component. `https://*.com` will not work.
+
+* The wildcard **MUST** be located in the subdomain furthest from the root domain. `https://sub.*.example.com` will not work.
+
+* The URL **must not** contain more than one wildcard. `https://*.*.example.com` will not work.
+
+* A wildcard **may** be prefixed and/or suffixed with additional valid hostname characters. `https://prefix-*-suffix.example.com` will work.  A wildcard for subdomains of cloud provider **should** always include at least a prefix or a suffix, such as `https://my-app-prefix-*.herokuapp.com`.
```

Comment:
A prefix on something like a `herokuapp.com` only gives an illusion of added security, but it's no more than that as anyone can create the `my-app-prefix-but-actually-an-evil-app` Heroku app.

For that reason I don't think we should endorse or suggest this by using it as an example and would rather see us use a fictitious domain name in our examples. Customers can still use this on Heroku but let them put 2 and 2 together...

Reply:
We provide this guidance, which is linked in the beginning of this document: https://auth0.com/docs/best-practices/application-settings#do-not-use-wildcards-or-localhost-in-callbacks-or-origins-fields  Do you think we should include something like that directly in this document as well?

I'll remove the cloud provider example for now.

Reply:
Good idea, perhaps  a note towards the top of the page (see my other comment)?